libpod/container_internal_linux.go

```diff
@@ -205,6 +205,27 @@ func (c *Container) resolveWorkDir() error {
 		// the path exists on the container.
 		if err != nil {
 			if os.IsNotExist(err) {
+
+				// We cannot create workdir since explicit --workdir is set in config but workdir could also be a symlink.
+				// If its a symlink lets check if resolved link is present on the container or not, If we can resolve symlink
+				// and resolved link is present on the container then return nil cause its a vaild use-case.
+				workDirInfo, err := os.Lstat(resolvedWorkdir)
+				if workDirInfo != nil && err == nil && (workDirInfo.Mode()&os.ModeSymlink == os.ModeSymlink) {
+					resolvedSymlink, err := os.Readlink(resolvedWorkdir)
+					if err == nil && resolvedSymlink != "" {
```

Comment:
could we simplify it by using directly `os.Readlink` instead of using `os.Lstat` first?

If `os.Readlink` succeeds we know it is a symlink and we don't need the stat.

Reply:
fixed